File: Ashutosh-Singh/SlidingWindow/Maximum-Erasure-Value.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int n = nums.size();
        int sum = 0;
        int maxSum = 0;
        unordered_set<int> uniqueSet;

        int left = 0;
        for (int right = 0; right < n; ++right) {
            while (uniqueSet.find(nums[right]) != uniqueSet.end()) {
                uniqueSet.erase(nums[left]);
                sum -= nums[left];
                ++left;
            }
            uniqueSet.insert(nums[right]);
            sum += nums[right];
            maxSum = max(maxSum, sum);
        }
        return maxSum;
    }
};

int main() {
    Solution solution;

    // Example input
    vector<int> nums = {4, 2, 4, 5, 6};

    // Calculate the maximum unique subarray sum
    int result = solution.maximumUniqueSubarray(nums);

    // Output the result
    cout << "Maximum Unique Subarray Sum: " << result << endl;

    return 0;
}

File: Anya_Seth/Sliding Window/Maximum Erasure Value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int start=0;
        int end=0;
        int max=0;
        int sum=0;
        for(int i=0;i<nums.length;i++){
            max=Math.max(max,nums[i]);
        }
        int[] freq=new int[max+1];
        int maxSum=0;
        while(end<nums.length){
            freq[nums[end]]++;
            sum+=nums[end];
            while(freq[nums[end]]>1){
                freq[nums[start]]--;
                sum-=nums[start];
                start++;
            }
            maxSum=Math.max(maxSum,sum);
            end++;
        }
        return maxSum;
    }
}
